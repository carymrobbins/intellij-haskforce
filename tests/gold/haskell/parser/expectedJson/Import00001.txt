[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":7,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":16}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Import1"]},null,null]},[],[{"importAnn":{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":48,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8}]},"importModule":{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},"srcInfoPoints":[]},"Prelude"]},"importAs":null,"importSrc":false,"importQualified":true,"importSpecs":{"ImportSpecList":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":48,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":26},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":32,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":26},{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":33},{"srcSpanFilename":"A.hs","srcSpanEndColumn":42,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":41},{"srcSpanFilename":"A.hs","srcSpanEndColumn":48,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":47}]},true,[{"IVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":41,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":34},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":41,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":34},"srcInfoPoints":[]},"zipWith"]}]},{"IVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":47,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":43},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":47,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":43},"srcInfoPoints":[]},"head"]}]}]]},"importPkg":null},{"importAnn":{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":38,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1}]},"importModule":{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Data.Maybe"]},"importAs":null,"importSrc":false,"importQualified":false,"importSpecs":{"ImportSpecList":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":38,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":19},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":19},{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":29},{"srcSpanFilename":"A.hs","srcSpanEndColumn":38,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":37}]},false,[{"IVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":20},"srcInfoPoints":[]},"fromMaybe"]}]},{"IVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":31},"srcInfoPoints":[]},"isJust"]}]}]]},"importPkg":null},{"importAnn":{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":8}]},"importModule":{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":18},"srcInfoPoints":[]},"Data.Maybe"]},"importAs":null,"importSrc":false,"importQualified":true,"importSpecs":{"ImportSpecList":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":29},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":29},{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":39}]},false,[{"IThingAll":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":39,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":30},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":35},{"srcSpanFilename":"A.hs","srcSpanEndColumn":38,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":36},{"srcSpanFilename":"A.hs","srcSpanEndColumn":39,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":38}]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":30},"srcInfoPoints":[]},"Maybe"]}]}]]},"importPkg":null},{"importAnn":{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":8},{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":26}]},"importModule":{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":18},"srcInfoPoints":[]},"Prelude"]},"importAs":{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":29},"srcInfoPoints":[]},"P"]},"importSrc":false,"importQualified":true,"importSpecs":null,"importPkg":null}],[]]},[]]