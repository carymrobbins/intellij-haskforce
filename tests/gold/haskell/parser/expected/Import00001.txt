Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Import1')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    IMPDECL
      PsiElement(HaskellTokenType.import)('import')
      PsiElement(HaskellTokenType.qualified)('qualified')
      QCONID
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('Prelude')
      PsiElement(HaskellTokenType.hiding)('hiding')
      PsiElement(HaskellTokenType.()('(')
      IMPORTT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('zipWith')
      PsiElement(HaskellTokenType.,)(',')
      IMPORTT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('head')
      PsiElement(HaskellTokenType.))(')')
    IMPDECL
      PsiElement(HaskellTokenType.import)('import')
      QCONID
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('Data')
        PsiElement(HaskellTokenType..)('.')
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('Maybe')
      PsiElement(HaskellTokenType.()('(')
      IMPORTT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('fromMaybe')
      PsiElement(HaskellTokenType.,)(',')
      IMPORTT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('isJust')
      PsiElement(HaskellTokenType.))(')')
    IMPDECL
      PsiElement(HaskellTokenType.import)('import')
      PsiElement(HaskellTokenType.qualified)('qualified')
      QCONID
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('Data')
        PsiElement(HaskellTokenType..)('.')
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('Maybe')
      PsiElement(HaskellTokenType.()('(')
      IMPORTT
        TYCON
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('Maybe')
        PsiElement(HaskellTokenType.()('(')
        PsiElement(HaskellTokenType...)('..')
        PsiElement(HaskellTokenType.))(')')
      PsiElement(HaskellTokenType.))(')')
    IMPDECL
      PsiElement(HaskellTokenType.import)('import')
      PsiElement(HaskellTokenType.qualified)('qualified')
      QCONID
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('Prelude')
      PsiElement(HaskellTokenType.as)('as')
      QCONID
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('P')